packages/model-viewer/src/model-viewer-base.ts

```diff
@@ -308,6 +324,12 @@ export default class ModelViewerElementBase extends UpdatingElement {
       // at all times:
       this[$isElementInViewport] = true;
     }
+    const ModelViewerGlobalConfig: ModelViewerGlobalConfig =
+        (self as any).ModelViewerElement || {};
+    if (ModelViewerGlobalConfig.powerPreference != null) {
+      ModelViewerElementBase.powerPreference =
+          ModelViewerGlobalConfig.powerPreference;
+    }
```

Comment:
So, after some testing, it turns out we can revert all the changes to this file. Your code is great and does what you want, but we're defeated by the browser. The powerPreference is only a preference, and it turns out if we create a 3D context on the high-performance GPU, then destroy it and create one with a "low-power" preference, the browser says "well, we're already using the high-performance GPU, so we don't need to bother switching just for a preference". So, it turns out we have to set powerPreference before the model-viewer library is loaded or it can't work at all, which we should make clear in the docs.

Reply:
Does that mean I should introduce a helper function like isDebugMode and retrieve the powerPreference that way?

Reply:
You can probably skip the helper, since unlike debug mode, we don't set this via URL query parameters.